pkg/proxy/ipvs/proxier.go

```diff
@@ -720,51 +720,58 @@ func (handle *LinuxKernelHandler) GetKernelVersion() (string, error) {
 }
 
 // CanUseIPVSProxier checks if we can use the ipvs Proxier.
-// This is determined by checking if all the required kernel modules can be loaded. It may
-// return an error if it fails to get the kernel modules information without error, in which
-// case it will also return false.
-func CanUseIPVSProxier(handle KernelHandler, ipsetver IPSetVersioner, scheduler string) error {
-	mods, err := handle.GetModules()
+// If the ipvs already has virtual servers (VS) we assume this is a re-start and
+// skip the tests. If the ipvs is empty we try to add a VS with the passed scheduler.
+// If that works we assume that ipvs and the passed scheduler are supported.
+// The ipset version is also checked.
+func CanUseIPVSProxier(ipvs utilipvs.Interface, ipsetver IPSetVersioner, scheduler string) error {
+	// Check is any virtual servers (vs) are configured. If any, we assume
+	// that this is a kube-proxy re-start and skip the checks.
+	vservers, err := ipvs.GetVirtualServers()
 	if err != nil {
-		return fmt.Errorf("error getting installed ipvs required kernel modules: %v", err)
+		klog.ErrorS(err, "Can't read the ipvs")
+		return err
 	}
-	loadModules := sets.NewString()
-	loadModules.Insert(mods...)
-
-	kernelVersionStr, err := handle.GetKernelVersion()
-	if err != nil {
-		return fmt.Errorf("error determining kernel version to find required kernel modules for ipvs support: %v", err)
+	klog.V(5).InfoS("Virtual Servers", "count", len(vservers))
+	if len(vservers) > 0 {
```

Comment:
This does not confirm the scheduler works - is that OK?  Why do we confirm that later and not here?

Reply:
My reasoning is that if virtual servers are configured in ipvs already, then kube-proxy has been succesfully added them earlier, so it is re-started, possibly upgraded, but no check of ipvs and the scheduler is needed.

That assumes that kube-proxy is the only used of ipvs, and that a change of the configured scheduled is a forbidden operation (at least not without a node reboot).

Reply:
What if the user changes the kube-proxy config to a new scheduler - is it fair to demand a reboot?  Will they know to do that?

Reply:
> What if the user changes the kube-proxy config to a new scheduler - is it fair to demand a reboot? Will they know to do that?

Good point. I have made an unnecessary assumption here. I will cover that and test.